src/core/highlight.ts

```diff
@@ -54,6 +54,10 @@ export function highlight (
 		language,
 	};
 
+	if (env.languageDef && !env.language) {
+		env.language = prism.languageRegistry.getLanguage(env.languageDef);
+	}
+
 	prism.hooks.run('before-tokenize', env);
```

Comment:
I thought the hook should have the full context during the run, so I decided to add this code before calling it. Am I right?